lib/Backend/Lower.cpp

```diff
@@ -4932,15 +4934,47 @@ Lowerer::LowerNewScObjArray(IR::Instr *newObjInstr)
         Assert(weakFuncRef);
     }
 
-    IR::Opnd *opndSrc1 = newObjInstr->UnlinkSrc1();
-    if (opndSrc1->IsImmediateOpnd())
+    IR::LabelInstr *labelDone = IR::LabelInstr::New(Js::OpCode::Label, func);
+    IR::Opnd *opndSrc = newObjInstr->GetSrc2();
+
+    Assert(opndSrc->IsSymOpnd());
+    StackSym *linkSym = opndSrc->AsSymOpnd()->m_sym->AsStackSym();
+    Assert(linkSym->IsSingleDef());
+    IR::Instr* argInstr = linkSym->GetInstrDef();
+    IR::Opnd *opndOfArrayCtor = argInstr->GetSrc1();
+    // Generate fast path only if it meets all the conditions:
+    // 1. 1st paramter is a constant
+    // 2. It is in range 0 <= parameter <= 8
+    // 3. It is the only parameter.
+    if (opndOfArrayCtor->IsIntConstOpnd() || opndOfArrayCtor->IsAddrOpnd()) // #1
```

Comment:
You can still call GetImmediateValue on the addr opnd, instead of this - in that way, you don't have to work around the fetching business.

---
In reply to: [71742831](https://github.com/Microsoft/ChakraCore/pull/1304#discussion_r71742831) [](ancestors = 71742831,71649822)